compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaLexer.java

```diff
@@ -1047,6 +1050,50 @@ private STToken processIdentifierOrKeyword() {
         }
     }
 
+    private int getNextNonWSOrNonCommentChar() {
+        int lookaheadCount = 0;
+        char nextChar = reader.peek(lookaheadCount);
+        while (nextChar != Character.MAX_VALUE) {
+            switch (nextChar) {
+                case LexerTerminals.SPACE:
+                case LexerTerminals.TAB:
+                case LexerTerminals.FORM_FEED:
+                case LexerTerminals.CARRIAGE_RETURN:
+                case LexerTerminals.NEWLINE:
+                    lookaheadCount++;
+                    break;
+                case LexerTerminals.SLASH:
+                    if (reader.peek(lookaheadCount + 1) == LexerTerminals.SLASH) {
+                        lookaheadCount += 2;
+                        lookaheadCount = skipComment(lookaheadCount);
+                        break;
+                    }
+                    return nextChar;
+                default:
+                    return nextChar;
+            }
+            nextChar = reader.peek(lookaheadCount);
+        }
+        return nextChar;
+    }
+
+    private int skipComment(int lookaheadCount) {
```

Comment:
@rdulmina Can we not reuse the same logic we used to skip similar cases in xml and string templates without using a new method?